### DISCUSSION TITLE
PhotoDemon 9.0 beta

### DISCUSSION THREAD

**Discussion by tannerhelland** [tannerhelland/PhotoDemon#421] [News]
After two years of work, a new PhotoDemon release is imminent. 

Please help me test version 9.0 before its official release (currently planned for **September 2022**).

A lengthy article highlighting all the new features and improvements is [available at photodemon.org](https://photodemon.org/2022/08/15/photodemon-9-0-beta.html).

Zip file checksums:
CRC-32: dfb61580
MD5: 5123c15c659a122285b21660da6bf8ff
SHA-1: 8fc0808b09eef57525bff65d56a4a997ef696c41
SHA-256: ab9316f1b18fd3dece517864959d831b241cf506304f9b9d505f473e044cafbf

<hr /><em>This discussion was created from the release <a href='https://github.com/tannerhelland/PhotoDemon/releases/tag/v9.0-beta.1'>PhotoDemon 9.0 beta</a>.</em>

**Comment by g-busch** [comment]
Congratulations on this milestone! I've been putting 9.0 Beta through its paces and haven't encountered any problems, at least with the tools and adjustments I use regularly. PD offers more resources than I could ever wish for in an image editor, and having little or no experience with the rest of them, it would take me forever to learn and test them all. Nevertheless, judging by the attention to details in this program's development, I wouldn't expect any significant problems to show up.

**Reply by tannerhelland** [reply]
🙏 Thank you for the kind words, @g-busch.  I appreciate you taking a look at the 9.0 beta!

**Comment by manfromarce** [comment]
Congrats for your amazing work as always!
There are just two minor issues I found:
- when you open an image, if the brush tool was preselected the image is immediately marked as changed (so it prompts for unsaved changes on exit, even if the brush wasn’t actually used)
- when you attempt to save changes on an image format which is only supported as input, it would be nice if PhotoDemon displayed the Save As dialog instead of an error message

I'll also check if there is something to improve in the italian translation as soon as I can.

**Reply by tannerhelland** [reply]
Two excellent bug reports, @manfromarce , thank you!  I apologize for these issues.  I have fixed both problems (I hope) in the latest nightly build, uploaded a few minutes ago.

Thank you also for checking the Italian translation.  I am very grateful for your help there 🙏

**Reply by manfromarce** [reply]
Thank you for fixing these so fast  @tannerhelland !
I've attached the updated Italian localization file.

However I found another minor issue: when trying to add an image as new layer (via drag-and-drop or paste), if the import fails the layer is not created but the Move tool is still automatically selected. 
In addition, since PhotoDemon supports pasting / dropping a file path string, it would be better to automatically recognize and remove double quotes, as Windows automatically adds them if you use context menu > Copy path.

[Italian.zip](https://github.com/tannerhelland/PhotoDemon/files/9426575/Italian.zip)

**Reply by tannerhelland** [reply]
Thank you so much, @manfromarce !  I am excited to have a complete Italian localization for this release.  

I have added your updated translation file and bug-fixes for your two issues to the latest nightly build.  Thank you for catching those bugs with the Move tool and `Windows Explorer > Copy as Path`.  (Honestly I had never tried the `Copy as Path` option before - I'm glad you discovered the double quotes issue!)  Similarly, the Move tool should no longer activate after a failed paste or drag+drop operation.

Many thanks 🙏

**Comment by chaes666** [comment]
Hope that Recent macros can add shortcut keys to quickly call Recent macros to improve work efficiency.

**Reply by tannerhelland** [reply]
Hello @chaes666 .  What shortcut keys would you suggest for the Recent Macros menu?

(I apologize that PhotoDemon does not currently support user-customizable hotkeys - though these are on the roadmap - but perhaps I can add a fixed hotkey for the most recent macro in the meantime?)

**Comment by chaes666** [comment]
It would be the best if you can add hotkeys to quickly call Recent macros. This way, when using Recent macros in a large number of images, you only need to use one hotkey to complete it, instead of searching in the menu, which increases the workload, thank you.
